ipatests/test_xmlrpc/test_host_plugin.py

```diff
@@ -337,6 +341,86 @@ def test_add_host_with_null_password(self, host):
+                **dict(
+                    setattr=u'ipamaxhostnamelength={}'.format(value)))
+            command()
+        except errors.EmptyModlist:
+            pass
+
+    @staticmethod
+    def generate_hostname(total_length):
+        if total_length < 9:
+            raise ArithmeticError("Total DNS length in theses tests"
+                                  "must be at least 9")
+        return "aaa.bbb.{}".format((total_length - 8) * "c")
+
+    def test_config_maxhostname_invalid(self, host):
+        """Change config maxhostname to an invalid value
+        (lower than MAXHOSTNAMELEN). Should fail"""
+        with raises_exact(errors.ValidationError(
+                name='ipamaxhostnamelength',
+                error='must be at least {}'.format(MAXHOSTNAMELEN))):
+            self.modify_config_maxhostname(host, MAXHOSTNAMELEN // 2)
+
+    def test_raise_hostname_limit_above_maxhostnamelen(self, host):
+        """Raise config maxhostname to a value above the default
+        (MAXHOSTNAMELEN). Should pass"""
+        self.modify_config_maxhostname(host, MAXHOSTNAMELEN * 2)
+
+    def test_try_hostname_length_above_maxhostnamelimit(self):
+        """Try to create host with hostname length above
+        hostnamelength limit. Should fail"""
+        testhost = HostTracker(name=u'testhost',
+                               fqdn=u'{}'.format(
+                                   self.generate_hostname(MAXHOSTNAMELEN + 1)))
+        self.modify_config_maxhostname(testhost, MAXHOSTNAMELEN)
+        with raises_exact(errors.ValidationError(
+                name=u'hostname',
+                error=u'can be at most {} characters'.format(
+                    MAXHOSTNAMELEN))):
+            testhost.create()
+            testhost.ensure_missing()
+
+    def test_try_hostname_length_below_maximum(self):
+        """Try to create host with valid hostname. Should pass"""
+        valid_length = MAXHOSTNAMELEN // 2
+        testhost = HostTracker(name=u'testhost',
+                               fqdn=u'{}'.format(
+                                   self.generate_hostname(valid_length)))
+        self.modify_config_maxhostname(testhost, MAXHOSTNAMELEN)
+        testhost.create()
+        testhost.ensure_missing()
+
+    def test_raise_limit_above_and_try_hostname_len_above_limit(self):
+        """Raise limit above default and try to create host with hostname
+        length above the new-set limit. Should fail"""
+        testhost = HostTracker(name=u'testhost',
+                               fqdn=u'{}'.format(
+                                   self.generate_hostname(MAXHOSTNAMELEN * 2)))
+        self.modify_config_maxhostname(testhost, MAXHOSTNAMELEN * 2)
+        with raises_exact(errors.ValidationError(
+                name=u'hostname',
+                error=u'invalid domain-name: DNS label cannot '
+                      u'be longer that {} characters'.format(
+                          MAXHOSTNAMELEN - 1))):
```

Comment:
If you implement the change I proposed above, the error msg should be related to 'hostname can be at most xx chars' and not 'DNS label'.